core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1183,7 +1183,7 @@ public static boolean isGlobalStateEquals(MetaData metaData1, MetaData metaData2
 
 
         public MetaDataDiff(MetaData before, MetaData after) {
-            uuid = after.uuid;
+            uuid = after.clusterUUID;
```

Comment:
I did. Side node - since it doesn't change I think we need an assertion  but that's a different change..